pkg/controller/runtime/adapter.go

```diff
@@ -46,7 +46,25 @@ type adapter struct {
 	watchFilterMu sync.Mutex
 }
 
-type watchFilter func(*resource.Metadata) bool
+type reducedMetadata struct {
```

Comment:
I think we can leave a note here, that `reducedMetadata` should always be comparable.

Reply:
updated